app/controllers/auth_controller.rb

```diff
@@ -30,6 +30,28 @@ def logout
     redirect_to auth_login_path
   end
 
+  before_action :set_user_by_session, only: [:change_password]
+
+  def change_password
+    return unless request.post?
+    if @user.authenticate(params[:current_password])
```

Comment:
I think the user should not be able to use the previous password as well...

what do you think?.

Reply:
The user can't do that, the use of the current password here is to ensure it's the user that is actually trying to change the password by inputting their original password. After that the user cannot use the old password.

Reply:
Actually they can...you can try it
let the new password be the same as the old password

it will go through.

Reply:
so you'd want to add a check like...

```
if params[:new_password] == params[:current_password]
    flash.now[:error] = "Choose a new password"
    render :change_password
end
```

Reply:
True true. I've made changes, the user can't use the old password anymore.

Reply:
it works perfectly now!

😄